### PR TITLE
Fix Python 3 serial driver issue.

### LINKED ISSUE #955: Serial driver python3 issues
```
edited line 173 in utilities.py to the following:
       cdc = next(serial.tools.list_ports.grep(name))
       return cdc[0]

and edited base_connection.py to add:
output += self.remote_conn.read(self.remote_conn.in_waiting).decode('utf-8', 'ignore')
on my line 385 (edited)
```

### PR DESCRIPTION
**Pull request by mreams13**
I made the changes that ktbyers specified in #955. I tested this with Python 3.6 and Windows 10. Fixes #955.

### CODE REVIEW

File: netmiko/utilities.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def check_serial_port(name):
     """returns valid COM Port."""
     try:
         cdc = next(serial.tools.list_ports.grep(name))
-        return cdc.split()[0]
+        return cdc[0]
```
Comment (ktbyers) [inline netmiko/utilities.py]: This change I don't understand as it implies either: 1)there was a bug before and it didn't work, 2)pyserial changed, 3)something else is going on.

What is the value of `cdc` here?

Comment (ktbyers) [reply netmiko/utilities.py]: I think someone else reported this to me at one point, but I didn't have a good way to test it (I am not sure if it was you or someone else).

Comment (ktbyers) [reply netmiko/utilities.py]: @mreams13

Comment (mreams13) [reply netmiko/utilities.py]: You opened the bug and suggested the fix. I think someone else might have suggested the bug on Slack. I am currently using netmiko over serial and I ran into the bug. I noticed it was still open with a fix included, but no one had actually changed it: 

"""
print(cdc) = COM6 - Prolific USB-to-Serial Comm Port (COM6)
print(cdc[0]) = COM6
print(cdc[1]) = Prolific USB-to-Serial Comm Port (COM6)
print(cdc[2]) = USB VID:PID=067B:2303 SER=5 LOCATION=1-6

ktbyers [8:14 PM]
Can you `print(type(cdc))`
"""

**Comment by ktbyers** [comment]
Can you fix the linting issues on the code? You can probably fix this by doing:

```
black .
pylama .
```

From the root of the Nornir project (i.e. the same directory as setup.py)

Note, pylama . might require that you do some additional modifications to the code.

**Pull request closed without merging**